src/AuthContext.tsx

```diff
@@ -68,17 +77,36 @@ export function AuthProvider({ children }: { children: React.ReactNode }) {
   useEffect(() => {
     const unsubscribe = onAuthStateChanged(auth, (user) => {
       setCurrentUser(user);
+
```

Comment:
the method looks good, but i think it would be better if you use async/await instead of .then(). also use try/catch blocks

Reply:
Fixed